dashboard/app/models/script.rb

```diff
@@ -741,6 +754,25 @@ def self.add_script(options, raw_stages)
     script
   end
 
+  # Clone this script, appending a dash and the suffix to the name of this
+  # script. Also clone all the levels in the script, appending an underscore and
+  # the suffix to the name of each level. Mark the new script as hidden, and
+  # copy any translations and other metadata associated with the original script.
+  def clone_with_suffix(new_suffix)
+    script_filename = "config/scripts/#{name}.script"
+    new_name = "#{name}-#{new_suffix}"
+    _, custom_i18n = Script.setup([script_filename], new_suffix: new_suffix)
+
+    # Omit any unused stage names. custom_i18n will already contain any stage
+    # still in use, and summarize_i18n(true) may return some stage names which
+    # are no longer in use.
+    metadata_i18n = summarize_i18n(false)
+    Script.merge_and_write_i18n(custom_i18n, new_name, metadata_i18n)
+
+    new_filename = "config/scripts/#{new_name}.script"
+    ScriptDSL.serialize(Script.find_by_name(new_name), new_filename)
```

Comment:
Nit: The script object is returned by `Script.setup`, no need to look it up again. I don't think the i18n stuff above would modify the script record itself.